crates/nu-cli/src/commands/to_xml.rs

```diff
@@ -0,0 +1,139 @@
+use crate::commands::WholeStreamCommand;
+use crate::prelude::*;
+use nu_errors::{ShellError};
+use nu_protocol::{Primitive, ReturnSuccess, Signature, SyntaxShape, UntaggedValue, Value};
+use std::io::Write;
+
+pub struct ToXML;
+
+#[derive(Deserialize)]
+pub struct ToXMLArgs {
+    pretty: Option<Value>,
+}
+
+#[async_trait]
+impl WholeStreamCommand for ToXML {
+    fn name(&self) -> &str {
+        "to xml"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("to xml").named(
+            "pretty",
+            SyntaxShape::Int,
+            "Formats the XML text with the provided indentation setting",
+            Some('p'),
+        )
+    }
+
+    fn usage(&self) -> &str {
+        "Convert table into .xml text"
+    }
+
+    async fn run(
+        &self,
+        args: CommandArgs,
+        registry: &CommandRegistry,
+    ) -> Result<OutputStream, ShellError> {
+        to_xml(args, registry).await
+    }
+}
+
+pub fn write_xml_events<W: Write>(value: &Value, writer: &mut xml::writer::EventWriter<W>) -> Result<(), ShellError> {
+    Ok(match &value.value {
+        UntaggedValue::Primitive(Primitive::Boolean(b)) => {
+            writer.write(b.to_string().as_str()).unwrap();
+        },
+        UntaggedValue::Row(o) => {
+            for (k, v) in o.entries.iter() {
+                writer.write(xml::writer::XmlEvent::start_element(k.as_str())).unwrap();
+                write_xml_events(v, writer).unwrap();
+                writer.write(xml::writer::XmlEvent::end_element()).unwrap();
+            }
+        },
+        UntaggedValue::Table(l) => {
+            writer.write(xml::writer::XmlEvent::start_element("table")).unwrap();
+            for value in l {
+                write_xml_events(value, writer).unwrap();
+            }
+            writer.write(xml::writer::XmlEvent::end_element()).unwrap();
+        },
+        _ => {
+            writer.write(xml::writer::XmlEvent::start_element("unknown")).unwrap();
+            writer.write(xml::writer::XmlEvent::end_element()).unwrap();
+        }
+    })
+}
+
+async fn to_xml(
+    args: CommandArgs,
+    registry: &CommandRegistry,
+) -> Result<OutputStream, ShellError> {
+    let registry = registry.clone();
+    let name_tag = args.call_info.name_tag.clone();
+    let (ToXMLArgs { pretty }, input) = args.process(&registry).await?;
+    let name_span = name_tag.span;
+    let input: Vec<Value> = input.collect().await;
+
+    let to_process_input = match input.len() {
+        x if x > 1 => {
+            let tag = input[0].tag.clone();
+            vec![Value {
+                value: UntaggedValue::Table(input),
+                tag,
+            }]
+        }
+        1 => input,
+        _ => vec![],
+    };
+
+    Ok(
+        futures::stream::iter(to_process_input.into_iter().map(move |value| {
+            let value_span = value.tag.span;
+
+            let indent = match &pretty {
```

Comment:
Can this be simplified?

Reply:
Yes, probably using `map_or` since you are just using 0 as a default
https://doc.rust-lang.org/std/option/enum.Option.html#method.map_or
https://doc.rust-lang.org/std/result/enum.Result.html#method.map_or

Reply:
Cool, I even discovered the helpful `p.value.expect_int()`, which smoothes things a bit more.